examples/nlp/md/text_extraction_with_bert.md

```diff
@@ -0,0 +1,540 @@
+__________________________________________________________________________________________________
+activation_8 (Activation)       (None, 384)          0           flatten_1[0][0]                  
+==================================================================================================
+Total params: 109,483,776
+Trainable params: 109,483,776
+Non-trainable params: 0
+__________________________________________________________________________________________________
+
+```
+</div>
+---
+## Create evaluation Callback
+
+This callback will compute the exact match score using the validation data
+after every epoch.
+
+
+
+```python
+
+def normalize_text(text):
+    text = text.lower()
+
+    # Remove punctuations
+    exclude = set(string.punctuation)
+    text = "".join(ch for ch in text if ch not in exclude)
+
+    # Remove articles
+    regex = re.compile(r"\b(a|an|the)\b", re.UNICODE)
+    text = re.sub(regex, " ", text)
+
+    # Remove extra white space
+    text = " ".join(text.split())
+    return text
+
+
+class ExactMatch(keras.callbacks.Callback):
+    """
+    Each `SquadExample` object contains the character level offsets for each token
+    in its input paragraph. We use them to get back the span of text corresponding
+    to the tokens between our predicted start and end tokens.
+    All the ground-truth answers are also present in each `SquadExample` object.
+    We calculate the percentage of data points where the span of text obtained
+    from model predictions matches one of the ground-truth answers.
+    """
+
+    def __init__(self, x_eval, y_eval):
+        self.x_eval = x_eval
+        self.y_eval = y_eval
+
+    def on_epoch_end(self, epoch, logs=None):
+        pred_start, pred_end = self.model.predict(self.x_eval)
+        count = 0
+        eval_examples_no_skip = [_ for _ in eval_squad_examples if _.skip == False]
+        for idx, (start, end) in enumerate(zip(pred_start, pred_end)):
+            squad_eg = eval_examples_no_skip[idx]
+            offsets = squad_eg.context_token_to_char
+            start = np.argmax(start)
+            end = np.argmax(end)
+            if start >= len(offsets):
+                continue
+            pred_char_start = offsets[start][0]
+            if end < len(offsets):
+                pred_char_end = offsets[end][1]
+                pred_ans = squad_eg.context[pred_char_start:pred_char_end]
+            else:
+                pred_ans = squad_eg.context[pred_char_start:]
+
+            normalized_pred_ans = normalize_text(pred_ans)
+            normalized_true_ans = [normalize_text(_) for _ in squad_eg.all_answers]
+            if normalized_pred_ans in normalized_true_ans:
+                count += 1
+        acc = count / len(self.y_eval[0])
+        print(f"\nepoch={epoch+1}, exact match score={acc:.2f}")
+
+
+```
+
+---
+## Train and Evaluate
+
+
+
+```python
+exact_match_callback = ExactMatch(x_eval, y_eval)
+model.fit(
+    x_train,
+    y_train,
+    epochs=1,
```

Comment:
Please add a comment here that the recommend number of epochs is 3, not 1. You don't need to regenerate the files, you can juste edit the ipynb and md files directly to add the comment.